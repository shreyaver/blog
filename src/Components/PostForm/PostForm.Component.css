.Form-content .Button {
    margin: 0 auto;
}
.Form-content {
    width: 70%;
    min-width: 300px;
    height: 47vw;
    min-height: 830px;
    margin: auto;
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    justify-content: space-evenly;
    color: rgb(115, 115, 115);
    text-transform: uppercase;
    font-size: 18px;
}
.Form-content span {
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    height: 40px;
}
.Form-content input, textarea {
    width: 40vw;
    max-width: 700px;
    min-width: 150px;
    height: 30px;
    font-size: 18px;
    color: grey;
    border: 1px solid rgb(216, 216, 216);
}
.Form-content label {
    display: inline-block;
    width: 40vw;
    max-width: 700px;
    min-width: 150px;
}
.Form-content .titleEntry {
    height: 50px;
}
.Form-content .descriptionEntry {
    height: 80px;
}
.Form-content input[type=text] {
    height: 100%;
}
.Form-content input[type=file] {
    height: 100%;
    border: none;
}
.Form-content textarea {
    height: 100%;
}