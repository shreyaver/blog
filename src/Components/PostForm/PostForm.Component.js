import React, { Component } from 'react';
import './PostForm.Component.css';
import Button from '../Button/Button.Component';
import Axios from 'axios';

class PostForm extends Component  {
    state = {
        date: '',
        readingTime: '',
        title: '',
        description: ''
    }
    handleInput = (changedField) => {
        this.setState({ [changedField.target.className]: changedField.target.value })
    }
    handleSubmit = (e) => {
        e.preventDefault();
        const form = new FormData();
        Object.keys(this.state).forEach((key) => form.append(key, this.state[key]));
        return Axios.post('/post', form).then((payload) => {
            console.log(payload.data);
            return payload.data;
        });
    }
    render() {
    return (
            <form className = "Form-content" encType = "multipart/form-data" onSubmit={this.handleSubmit}> 
            <span>
                <label>
                    Date 
                </label>
                <input type = "date" onChange = {this.handleInput} value = {this.state.date} className = "date" />
            </span>
            <span>
                <label>
                    Reading time (in minutes)
                </label>
                <input type = "number" onChange = {this.handleInput} className = "readingTime" />
            </span>
            <span className = "titleEntry">
                <label>
                    Title (Characters: {70-this.state.title.length})
                </label>
                <input type = "text" maxLength = {70} onChange = {this.handleInput} value = {this.state.title} className = "title"/>
            </span>
            <span className = "descriptionEntry">
                <label>
                    Description (Characters: {130-this.state.description.length})
                </label>
                <textarea maxLength = {130} onChange = {this.handleInput} value = {this.state.description} className = "description"/>
            </span>
            <span>
                <label> 
                    Image
                </label>
                <input type="file" name="image" accept="image/*"></input>
            </span>
            <span>
                <Button type = "submit" caption = "UPLOAD POST" /> 
            </span>
            </form>
    )}
}
export default PostForm;