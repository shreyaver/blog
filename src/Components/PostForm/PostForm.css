.Form-content .Button {
    margin: 0 auto;
}
.Form-content {
    width: 70%;
    min-width: 300px;
    height: 47vw;
    min-height: 830px;
    margin: auto;
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    justify-content: space-evenly;
    color: rgb(115, 115, 115);
    text-transform: uppercase;
    font-size: 18px;
}
.Form-content span {
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    height: 40px;
}
.Form-content input, textarea {
    width: 40vw;
    max-width: 700px;
    min-width: 150px;
    height: 30px;
    font-size: 18px;
    color: grey;
}
.Form-content label {
    display: inline-block;
    width: 40vw;
    max-width: 700px;
    min-width: 150px;
}
.Form-content .title {
    height: 50px;
}
.Form-content .description {
    height: 120px;
}
.Form-content input[type=text] {
    height: 100%;
}
.Form-content textarea {
    height: 100%;
}