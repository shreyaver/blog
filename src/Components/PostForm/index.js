import React from 'react';
import './PostForm.css';
import Button from '../Button';

const PostForm = (props) => {
    return (
            <form className = "Form-content">
            <span>
                <label>
                    Date 
                </label>
                <input type = "date" />
            </span>
            <span>
                <label>
                    Reading time (in minutes)
                </label>
                <input type = "number"/>
            </span>
            <span className = "title">
                <label>
                    Title 
                </label>
                <input type = "text" maxLength = {70}/>
            </span>
            <span className = "description">
                <label>
                    Description 
                </label>
                <textarea maxLength = {130}/>
            </span>
            <span>
                <label> 
                    Image
                </label>
                <input type="file" name="pic" accept="image/*"></input>
            </span>
            <span>
                <Button type = "submit" caption = "UPLOAD POST" />
            </span>
            </form>
    )
}
export default PostForm;