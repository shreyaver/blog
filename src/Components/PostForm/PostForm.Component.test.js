import React from 'react';
import renderer from 'react-test-renderer';
import PostForm from './PostForm.Component.js';
import { shallow } from 'enzyme';
import axios from 'axios';

describe('PostForm', () => {
  const wrapper = shallow(<PostForm />);
  const cardData = {
    "date": "2nd Januray, 2018",
    "readingTime": "2 mins",
    "title": "The future of abstract art and the culture ...",
    "description": "Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...",
    "claps": 10,
    "liked": false,
    "image": "abstract.png"
  }
  let postMock;
  beforeAll(() => {
    postMock = jest.spyOn(axios, 'post');
    postMock.mockImplementation(() => {
      return Promise.resolve({ 'data': cardData })
    });
  });
  afterAll(() => {
    postMock.mockRestore();
  });
  it('renders without crashing', () => {
    const postForm = renderer.create(<PostForm />).toJSON();
    expect(postForm).toMatchSnapshot();
  });
  it('sets state to entered input', () => {
    wrapper.find('.title').simulate('change', { target: { className: 'title', value: 'ball' } });
    expect(wrapper.instance().state.title).toEqual('ball');
  });
  it('calls "handleSubmit()" on submitting form',() => {
    const mockFn=jest.fn();
    wrapper.instance().handleSubmit=mockFn();
    wrapper.find('.Form-content').simulate('submit', { target: { className: 'Form-content'}, preventDefault: jest.fn() });
    expect(mockFn).toHaveBeenCalled();
  });
  it('sends post request and receives response via "handleSubmit()"', async (done) => {
    const newWrapper = shallow(<PostForm />);
    expect(await newWrapper.instance().handleSubmit({preventDefault: jest.fn()} )).toEqual(cardData);
    done();
  });
});