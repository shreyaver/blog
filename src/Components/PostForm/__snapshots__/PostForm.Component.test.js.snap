// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostForm renders without crashing 1`] = `
<form
  className="Form-content"
  encType="multipart/form-data"
  onSubmit={[Function]}
>
  <span>
    <label>
      Date
    </label>
    <input
      className="date"
      onChange={[Function]}
      type="date"
      value=""
    />
  </span>
  <span>
    <label>
      Reading time (in minutes)
    </label>
    <input
      className="readingTime"
      onChange={[Function]}
      type="number"
    />
  </span>
  <span
    className="titleEntry"
  >
    <label>
      Title (Characters: 
      70
      )
    </label>
    <input
      className="title"
      maxLength={70}
      onChange={[Function]}
      type="text"
      value=""
    />
  </span>
  <span
    className="descriptionEntry"
  >
    <label>
      Description (Characters: 
      130
      )
    </label>
    <textarea
      className="description"
      maxLength={130}
      onChange={[Function]}
      value=""
    />
  </span>
  <span>
    <label>
      Image
    </label>
    <input
      accept="image/*"
      name="image"
      type="file"
    />
  </span>
  <span>
    <button
      className="Button"
      type="submit"
    >
      UPLOAD POST
    </button>
  </span>
</form>
`;
